lmdeploy/pytorch/chat.py

```diff
@@ -1,158 +1,153 @@
-                      'use plain huggingface model.')
-
-        model = model.cuda(local_rank)
-    else:
-        config = dict(
-            tensor_parallel=dict(tp_size=world_size),  # Number of GPU
-            dtype=torch.float16,  # dtype of the weights (fp16)
-            replace_with_kernel_inject=True,
-            # replace the model with the kernel injector
-            max_out_tokens=2048,
-        )
-        # For internlm model not supported by DeepSpeed,
-        # set replace_with_kernel_inject=False to use AutoTP.
-        # It's a hotfix before the progress is merged
-        # https://github.com/InternLM/lmdeploy/issues/136
-        if 'InternLM' in model.__class__.__name__:
-            try:
-                # Use customized deepspeed supporting internlm
-                # https://github.com/wangruohui/DeepSpeed/tree/support_internlm_0.10.0 (commit cdef2ce)  # noqa: E501
-                from deepspeed.module_inject.containers.internlm import \
-                    InternLMLayerPolicy
-            except ImportError:
-                # use stock deepspeed
-                config.update({'replace_with_kernel_inject': False})
-            else:
-                for module in model.modules():
-                    if module.__class__.__name__ == 'InternLMDecoderLayer':
-                        InternLMLayerPolicy._orig_layer_class = module.__class__  # noqa: E501
-                        break
-
-        model = deepspeed.init_inference(
-            model=model,  # Transformers models
-            config=config,
-        )
-
-    return tokenizer, model
+- `exit`: terminate and exit chat
+- `config set key=value`: change generation config `key` to `value`,
+    e.g. config temperature=0 disable sampling for following chats
+- `clear`: clear chat history
+"""
+
+import itertools
+import logging
+from typing import Optional
+
+import fire
+import torch
+from transformers import GenerationConfig, PreTrainedModel
+
+from .adapters import init_adapter
+from .dist import get_local_rank, get_rank, get_world_size
+from .model import accel_model, init_model
+from .session import BasicSessionManager  # noqa: F401
+from .session import BasicSessionManagerWithHistory
+from .utils import BasicStreamer, TerminalIO, control
+
+logger = logging.getLogger(__name__)
+
+
+def set_logging(log_file, debug: bool):
+    torch.set_printoptions(linewidth=120)
+    level = logging.DEBUG if debug else logging.INFO
+    log_file = log_file or 'chat.log'
+    if r := get_rank() != 0:
+        log_file = log_file + f'.{r}'
+    logging.basicConfig(level=level,
+                        format=('%(filename)s: '
+                                '%(levelname)s: '
+                                '%(funcName)s(): '
+                                '%(lineno)d:\t'
+                                '%(message)s'),
+                        filename=log_file,
+                        filemode='w')
+    print(f'Worker {get_rank()} logging to {log_file}')
```

Comment:
why not use a logger?

Reply:
switch screen output into a file. A user can redirect it by `>`

Reply:
But user need to input prompt to the model and receive model's response on screen. If we use `>`, to redirect everything, user have to read model's response in files. This breaks the function of this program. 

If we just consider this print, I can change to use the logger in turbomind/utils.py as Han suggest, which is mmlogger internally. But mmlogger log only to master rank. This is also not perfect for this information. 

Maybe reconsider after rebased to #188.